gem/lib/pupperware/spec_helper.rb

```diff
@@ -236,6 +255,38 @@ def docker_volume_cp(src_path:, dest_volume:, dest_dir:, is_compose: true, uid:,
     end
   end
 
+  # takes a given certname, and generates certs to the dest_dir of dest_volume
+  # uses a transient pe-client-tools container with ssl.sh to generate certs
+  def generate_certificate(ca_service_name:, certname:, dns_alt_names:, dest_volume:, dest_dir:, is_compose: true, uid:, gid: )
+    uid ||= 'root'
+    gid ||= 'root'
+    if is_compose
+      prefix = ENV['COMPOSE_PROJECT_NAME'] || File.basename(Dir.pwd)
+      dest_volume = "#{prefix}_#{dest_volume}"
+    end
+    # create a temp container that generates certs to the appropriate volume
+    ca_container = get_service_container(ca_service_name)
+    ssl_src_path = Pathname.new(File.join(__dir__, 'compose-services', 'pe-postgres-custom'))
+    cmd = "docker run \
+      --rm \
+      --network #{get_container_network(ca_container)} \
+      --volume #{ssl_src_path}:/tmp/ssl \
+      --volume #{dest_volume}:/opt \
+      --entrypoint /bin/sh \
+      alpine/openssl \
```

Comment:
Found out that there's a small `alpine/openssl` that can be used here for this use case